core/src/main/java/com/dtolabs/rundeck/core/utils/ScriptExecUtil.java

```diff
@@ -248,14 +289,24 @@ public static int runLocalCommand(
             }
             return result;
         } catch (InterruptedException e) {
-            exec.destroy();
+            if (null != killHandler) {
+                killHandler.accept(exec.toHandle());
+            }
             if(exec.isAlive()){
                 exec.waitFor();
             }
             throw new InterruptedException("Execution interrupted with code: " + exec.exitValue());
         }
     }
 
+    public static void killProcessHandleDescend(ProcessHandle handle) {
+        handle.descendants().forEach(ScriptExecUtil::killProcessHandleDescend);
```

Comment:
As discussed, just calling `ScriptExecUtil::killProcessHandle` should be enough here as `descendants()` already return children recursively.